test/basics.jl

```diff
@@ -0,0 +1,60 @@
+test_string_length = 100
+
+#string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str, UniStr]
+#string_types = [ASCIIStr, LatinStr, UTF8Str, UCS2Str, UTF16Str, UTF32Str]
+string_types = [ASCIIStr, LatinStr]
+
+##  create type specific test strings
+test_strings_base = Dict()
+for string_type in string_types
+    tmax = UInt32(typemax(Strs.codepoint_type(string_type)))
```

Comment:
this will have problems with the ones that could have surrogates.
Look at the code for `rand` of type `Char`.
I still need to add `rand` for all the CodePoint types, however `rand` has been in flux on master.